plugins/kn-image/README.md

```diff
@@ -0,0 +1,220 @@
+
+```
+apiVersion: v1
+data:
+  password: <docker_password>
+  username: <docker_username>
+kind: Secret
+metadata:
+  annotations:
+    tekton.dev/docker-0: us.icr.io
+  name: ibm-cr-push-secret
+  namespace: default
+type: kubernetes.io/basic-auth
+```
+Run it with:
+```
+kubetl create -f ImagePushSecret.yaml
+```
+### Secret for pulling image
+Besides, if your cluster account is not same with the account of container registry. Then maybe you also need to prepare another secret for pulling an image. Create a pull image secret called `ibm-cr-pull-secret`.
+```
+kubectl create secret docker-registry ibm-cr-pull-secret --docker-server=us.icr.io --docker-username=iamapikey --docker-password=${apikey} --docker-email=me@here.com
+```
+Note: docker-password is the apikey of the account of container registry
+
+## Create a ServiceAccount
+Create a seviceaccount yaml file named `sa.yaml`
+```
+apiVersion: v1
+imagePullSecrets:
+- name: ibm-cr-pull-secret
+kind: ServiceAccount
+metadata:
+  name: pipeline-account
+  namespace: default
+secrets:
+- name: ibm-cr-push-secret
+```
+Run it with:
+```
+kubectl create -f sa.yaml
+```
+
+## Deploy Knative service by using buildpacks builder task
+You can use `deploy` command line interface to deploy the image.
+```
+  kn-image deploy ${image_name} \
+    --builder buildpacks-v3 \
+    --git-url ${url_of_git_repo} \
+    --git-revision ${revision_of_git_repo} \
+    --saved-image ${generated_image_path} \
+    --serviceaccount ${ServiceAccount_for_ pushing_image} \
+    --namespcae ${namespace} --force
+```
+Now, we have 8 parameters, they are:
+- `--image_name`: generated image name
+- `--builder`: a builder can be used to build image
+- `--git-url`: the url of git repo
+- `--git-revision`: revision of the relative git repo
+- `--saved-image`: generated image path
+- `--serviceaccount`: a ServiceAccount for pushing image
+- `--namespace`: a namespace for building images. If you don't specify, it is default
+- `--force`: create service forcefully, replaces existing service if any. Default is false
+
+## Redeploy Knative service
+Sometimes, we want to modify some parameters and redeploy this image. Then we can use `redeploy` CLI to achieve our targets.
+For example, we only want to change the git-revision, then we can simply use below command to come true:
+```
+  kn-image redeploy ${image_name} \
+    --saved-image ${new_path_of_generated_image} \
+    --git-revision ${new-revision} \
+    --namespcae ${namespace}
+```
+
+## Deploy Knative service by kaniko builder task
+You can also use `kaniko` builder to deploy images, all steps are same with `buildpacks`. Below is an example about the deployment using kaniko:
+```
+  kn-image deploy kanikotest \
+    --builder kaniko \
+    --git-url https://github.com/bluebosh/knap-example \
+    --saved-image us.icr.io/knative_jordan/kanikotest:latest \
+    --serviceaccount default \
+    --force
+```
+
+## Example
```

Comment:
I’d separate this into a section called “Example Workflow” or similar...

Reply:
Change the name from `Example ` to `Example Workflow`